src/layers/components/ProductPhotos/ProductPhotos.ts

```diff
@@ -0,0 +1,85 @@
+import Observer from '../../Observer/Observer';
+import { IProductPhotos } from './ProductPhotos.types';
+import './ProductPhotos.scss';
+import { EventName } from '../../Observer/Observer.types';
+import Drift from 'drift-zoom';
+import { IModelData } from '../../Model/Model.types';
+
+class ProductPhotos {
+    private container: HTMLElement;
+    private observer: Observer;
+    private currImg?: string;
+    private data: IModelData;
+
+    constructor({ container, observer, currImg, data }: IProductPhotos) {
+        this.container = container;
+        this.observer = observer;
+        this.currImg = currImg;
+        this.data = data;
+    }
+
+    public render(): void {
+        const productPhotos = document.createElement('div');
+        productPhotos.classList.add('product__photos');
+        this.container.prepend(productPhotos);
+
+        const productSlider = document.createElement('div');
+        productSlider.classList.add('product__slider');
+        productPhotos.append(productSlider);
+
+        const productSlides = this.data.currProduct.images;
+        productSlides.map((item) => {
+            const slide = document.createElement('img');
+            slide.classList.add('product__slide');
+            slide.src = `${item}`;
+            productSlider.append(slide);
+        });
+
+        productSlides.forEach(() => {
+            const slides = document.querySelectorAll('.product__slide');
+
+            slides.forEach((slide) => {
+                slide.addEventListener('click', (e: Event) => {
+                    const eventObject = { eventName: EventName.clickImg, eventPayload: e };
+                    this.observer.notify(eventObject);
+                });
+            });
+        });
+    }
+
+    public renderThumbnail(): void {
+        const productThumbnail = document.createElement('div');
```

Comment:
You can create separate function createElementWithClass (for example) that will create element and add class to it instead of repeating these lines:
`const slide = document.createElement('img');
            slide.classList.add('product__slide');`

Reply:
Переделал в коммите [9ad0282](https://github.com/llallonen/online-store/pull/65/commits/9ad028238e34d454138a8fd20f56ad989c91b5b2)